treatment-service/models/storage.go

```diff
@@ -238,6 +234,22 @@ func (i *ExperimentIndex) isActive() bool {
 	return (i.StartTime.Before(time.Now()) || i.StartTime.Equal(time.Now())) && i.EndTime.After(time.Now())
 }
 
+func (i *ExperimentIndex) checkSegmentHasWeakMatch(segmentName string) bool {
+	if set, exists := i.stringSets[segmentName]; !exists || set.Len() == 0 {
```

Comment:
By this logic, int / real / bool segments will always return `true`. We should reverse the checks instead, like:

```golang
    if set, exists := i.stringSets[segmentName]; exists && set.Len() > 0 {
        return false
    }
    if ...
    return true
```

Better yet:

```golang
    if set, exists := i.stringSets[segmentName]; exists {
        if set.Len() > 0 {
                return false
        }
    } else if set, exists := i.intSets[segmentName]; exists {
        ...
    }
    ...
    return true
```

Reply:
Ooops logic failure there 😅 Thanks for catching this!

Reply:
Alright, just fixed this with your suggestion! :)